ui/plugins/auto_classification/controller.js

```diff
@@ -1,372 +1,924 @@
 "use strict";
 
-treeherder.controller('ClassificationPluginCtrl', [
-    '$scope', '$rootScope', 'ThLog', 'ThFailureLinesModel', 'ThClassifiedFailuresModel',
-    'ThMatcherModel', '$q', 'thEvents', 'thTabs', '$timeout', 'thNotify',
-    function ClassificationPluginCtrl(
-        $scope, $rootScope, ThLog, ThFailureLinesModel, ThClassifiedFailuresModel,
-        ThMatcherModel, $q, thEvents, thTabs, $timeout, thNotify) {
-        var $log = new ThLog(this.constructor.name);
+// TODO: move these into services or something
```

Comment:
Seems like this would be good.  This file has lots of different pieces.  Perhaps they could benefit from some logical separation.  :)